master/buildbot/test/unit/test_reporter_gerrit.py

```diff
@@ -70,6 +85,29 @@ def testSummaryCB(buildInfoList, results, status, arg):
                             (GERRIT_LABEL_VERIFIED, verified))
 
 
+@defer.inlineCallbacks
+def testSummaryCBDeferred(buildInfoList, results, master, arg):
+    success = False
+    failure = False
+
+    for buildInfo in buildInfoList:
+        yield  # placeholder for something like utils.getDetailsForBuild(master, buildInfo['build'], wantSteps=True)
+        if buildInfo['result'] == SUCCESS:
+            success = True
+        else:
+            failure = True
+
+    if failure:
+        verified = -1
+    elif success:
+        verified = 1
+    else:
+        verified = 0
+
+    defer.returnValue(makeReviewResult(str(buildInfoList),
```

Comment:
You should rather do:

    result = yield makeReviewResult(...)
    self.expect...(results[..], foo)